### ISSUE TITLE
build fails with musl libc 1.2

### ISSUE THREAD

**Issue by ncopa** [Leont/crypt-rijndael#17]
Build fails when building on alpine linux with musl 1.2 on 32 bit architectures (armv6, armv7 and x86). I have not tested 64 bit architectures yet.

```
Running Mkbootstrap for Rijndael ()
cc -c   -D_REENTRANT -D_GNU_SOURCE -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Os -fomit-frame-pointer   -DVERSION=\"1.14\" -DXS_VERSION=\"1.14\" -fPIC "-I/usr/lib/perl5/core_perl/CORE"   _rijndael.c
cc -c   -D_REENTRANT -D_GNU_SOURCE -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Os -fomit-frame-pointer   -DVERSION=\"1.14\" -DXS_VERSION=\"1.14\" -fPIC "-I/usr/lib/perl5/core_perl/CORE"   Rijndael.c
chmod 644 "Rijndael.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Rijndael.bs blib/arch/auto/Crypt/Rijndael/Rijndael.bs 644
Skip blib/lib/Crypt/Rijndael.pm (unchanged)                                    
In file included from _rijndael.c:21:
rijndael.h:85:10: error: unknown type name '__uint32_t'                        
   85 |  typedef __uint32_t UINT32;
      |          ^~~~~~~~~~
rijndael.h:86:10: error: unknown type name '__uint8_t'                         
   86 |  typedef __uint8_t  UINT8;
      |          ^~~~~~~~~
make: *** [Makefile:341: _rijndael.o] Error 1                                  
make: *** Waiting for unfinished jobs....                                      
In file included from Rijndael.xs:28:
ppport.h:4476: warning: "WIDEST_UTYPE" redefined                               
 4476 | # define WIDEST_UTYPE U32
      |
In file included from /usr/lib/perl5/core_perl/CORE/perl.h:2508,               
                 from Rijndael.xs:25:
/usr/lib/perl5/core_perl/CORE/handy.h:1072: note: this is the location of the previous definition
 1072 | #   define WIDEST_UTYPE U64
      |
```

**Comment by mjeveritt** [comment]
Fails for me with amd64/x86_64 musl 1.2.0 also. Thanks for fix.

**Comment by mjeveritt** [comment]
Success! :smiley: 
```
* Package:    dev-perl/Crypt-Rijndael-1.140.0
 * Repository: gentoo
 * Maintainer: perl@gentoo.org
 * USE:        abi_x86_64 amd64 elibc_musl kernel_linux userland_GNU
 * FEATURES:   distcc network-sandbox preserve-libs sandbox userpriv usersandbox
>>> cfg-update-1.8.2-r1: Creating checksum index...
>>> Unpacking source...
>>> Unpacking Crypt-Rijndael-1.14.tar.gz to /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/work
>>> Source unpacked in /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/work
>>> Preparing source in /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/work/Crypt-Rijndael-1.14 ...
 * Applying 4aa8da5835cd7d71efaf4ad126381626dcba49b7.patch ...
 [ ok ]
 * User patches applied.
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/work/Crypt-Rijndael-1.14 ...
 * Using ExtUtils::MakeMaker
 * perl Makefile.PL PREFIX=/usr INSTALLDIRS=vendor INSTALLMAN3DIR=none DESTDIR=/var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/image/
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Crypt::Rijndael
Writing MYMETA.yml and MYMETA.json
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/work/Crypt-Rijndael-1.14 ...
 * emake OTHERLDFLAGS=-Wl,-O1 -Wl,--as-needed
make -j6 'OTHERLDFLAGS=-Wl,-O1 -Wl,--as-needed' 
Running Mkbootstrap for Rijndael ()
x86_64-gentoo-linux-musl-gcc -c   -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -pipe -mtune=bonnell   -DVERSION=\"1.14\" -DXS_VERSION=\"1.14\" -fPIC "-I/usr/lib/perl5/5.30.3/x86_64-linux/CORE"   _rijndael.c
"/usr/bin/perl" "/usr/lib/perl5/5.30.3/ExtUtils/xsubpp"  -typemap '/usr/lib/perl5/5.30.3/ExtUtils/typemap' -typemap '/var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/work/Crypt-Rijndael-1.14/typemap'  Rijndael.xs > Rijndael.xsc
chmod 644 "Rijndael.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Rijndael.bs blib/arch/auto/Crypt/Rijndael/Rijndael.bs 644
cp Rijndael.pm blib/lib/Crypt/Rijndael.pm
mv Rijndael.xsc Rijndael.c
x86_64-gentoo-linux-musl-gcc -c   -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -pipe -mtune=bonnell   -DVERSION=\"1.14\" -DXS_VERSION=\"1.14\" -fPIC "-I/usr/lib/perl5/5.30.3/x86_64-linux/CORE"   Rijndael.c
In file included from Rijndael.xs:28:
ppport.h:4471: warning: "WIDEST_UTYPE" redefined
 4471 | #  define WIDEST_UTYPE U64TYPE
      | 
In file included from /usr/lib/perl5/5.30.3/x86_64-linux/CORE/perl.h:2510,
                 from Rijndael.xs:25:
/usr/lib/perl5/5.30.3/x86_64-linux/CORE/handy.h:1072: note: this is the location of the previous definition
 1072 | #   define WIDEST_UTYPE U64
      | 
Rijndael.xs: In function ‘XS_Crypt__Rijndael_set_iv’:
Rijndael.xs:96:16: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘STRLEN’ {aka ‘long unsigned int’} [-Wformat=]
rm -f blib/arch/auto/Crypt/Rijndael/Rijndael.so
x86_64-gentoo-linux-musl-gcc  -shared -O2 -pipe -mtune=bonnell -Wl,-O1 -Wl,--as-needed  _rijndael.o Rijndael.o -Wl,-O1 -Wl,--as-needed -o blib/arch/auto/Crypt/Rijndael/Rijndael.so  \
      \
  
chmod 755 blib/arch/auto/Crypt/Rijndael/Rijndael.so
>>> Source compiled.
 * Skipping make test/check due to ebuild restriction.
>>> Test phase [disabled because of RESTRICT=test]: dev-perl/Crypt-Rijndael-1.140.0

>>> Install dev-perl/Crypt-Rijndael-1.140.0 into /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/image/
make -j6 pure_install 
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Rijndael.bs blib/arch/auto/Crypt/Rijndael/Rijndael.bs 644
Files found in blib/arch: installing files in blib/lib into architecture dependent library tree
Installing /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/image/usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/Rijndael.so
Installing /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/image/usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/Crypt/Rijndael.pm
 * Fixing packlist file /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/.packlist
>>> Completed installing dev-perl/Crypt-Rijndael-1.140.0 into /var/tmp/portage/dev-perl/Crypt-Rijndael-1.140.0/image/

 * Final size of build directory: 488 KiB
 * Final size of installed tree:  100 KiB

strip: x86_64-gentoo-linux-musl-strip --strip-unneeded -N __gentoo_check_ldflags__ -R .comment -R .GCC.command.line -R .note.gnu.gold-version
   /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/Rijndael.so
./
./usr/
./usr/share/
./usr/share/doc/
./usr/share/doc/Crypt-Rijndael-1.140.0/
./usr/share/doc/Crypt-Rijndael-1.140.0/Changes.bz2
./usr/share/doc/Crypt-Rijndael-1.140.0/README.bz2
./usr/lib/
./usr/lib/perl5/
./usr/lib/perl5/vendor_perl/
./usr/lib/perl5/vendor_perl/5.30.3/
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/Rijndael.so
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/.packlist
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/Crypt/
./usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/Crypt/Rijndael.pm
>>> Done.
 * checking 5 files for package collisions
>>> Merging dev-perl/Crypt-Rijndael-1.140.0 to /
--- /usr/
--- /usr/lib/
--- /usr/lib/perl5/
--- /usr/lib/perl5/vendor_perl/
--- /usr/lib/perl5/vendor_perl/5.30.3/
--- /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/
--- /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/Crypt/
>>> /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/Crypt/Rijndael.pm
--- /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/
--- /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/
>>> /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/
>>> /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/.packlist
>>> /usr/lib/perl5/vendor_perl/5.30.3/x86_64-linux/auto/Crypt/Rijndael/Rijndael.so
--- /usr/share/
--- /usr/share/doc/
>>> /usr/share/doc/Crypt-Rijndael-1.140.0/
>>> /usr/share/doc/Crypt-Rijndael-1.140.0/README.bz2
>>> /usr/share/doc/Crypt-Rijndael-1.140.0/Changes.bz2
>>> dev-perl/Crypt-Rijndael-1.140.0 merged.
>>> Regenerating /etc/ld.so.cache...
```
